client/my-sites/plan-features-2023-grid/components/plan-comparison-grid.tsx

```diff
@@ -571,7 +575,11 @@ const PlanComparisonGridFeatureGroupRowCell: React.FunctionComponent< {
 				( feature ) => feature.getSlug() === featureSlug
 		  )
 		: false;
-	const storageOption = gridPlan.features.storageOptions.find( ( option ) => ! option.isAddOn );
+	const storageOptions = gridPlan.features.storageOptions;
+	const defaultStorageOption = storageOptions.find( ( option ) => ! option.isAddOn );
+	const canUpgradeStorageForPlan =
```

Comment:
Non-blocker: If we have to derive `defaultStorageOption` and `canUpgradeStorageForPlan` with such a complexity, it's a strong signal that the underlying data structure doesn't represent the semantics we need too well. 

I'm aware that we have the architectural design discussion happening in p2-p4TIVU-aKR, so just wanted to raise that we should expect a greatly simplified version of these from the new architecture :)

Reply:
Totally agree here